fairscale/utils/parallel.py

```diff
@@ -58,18 +58,45 @@ def enable_pytorch_sync_bn(module: torch.nn.Module) -> None:
             layer._specify_ddp_gpu_num(1)  # type: ignore
 
 
-def get_global_group() -> None:
+def get_process_group_cached(ranks: Optional[List[int]] = None) -> ProcessGroup:
     """
-    Singleton PyTorch distributed group.
-    Inspired by https://github.com/pytorch/fairseq
+    Singleton PyTorch distributed group cache. Inspired by the code from fairseq.
 
-    For FSDP, it is important to use a global group, otherwise, inner FSDP instances
-    will not share the gradient reduction bucket buffer with the root instance, end up using
-    more GPU memory.
+    For FSDP, it is important to use the same group between outer and inner FSDP instances,
+    otherwise, inner FSDP instances will not share the gradient reduction bucket buffer with
+    the root instance, end up using more GPU memory.
```

Comment:
perhaps we can replace ", end up using more GPU memory" with a new sentence such as "This will result in increased GPU memory utilization" or something along those lines? There are multiple sentences separated by commas which makes this slightly confusing.